types/index.d.ts

```diff
@@ -31,52 +34,100 @@ interface AssembleOptions {
-    // TODO fingerprint(algorithm)
-    fromBuffer(buffer: Buffer, resolver: any, noCheck: boolean): Type; // TODO
-    // TODO fromString(str)
-    // TODO inspect()
-    // TODO isValid(val, opts)
-    // TODO random()
-    // TODO schema(opts)
-    toBuffer(value: object): Buffer;
-    // TODO toJSON()
-    // TODO toString(val)
-    // TODO wrap(val)
-  }
-
-  function forSchema(schema: Schema): Type;
-  function forValue(value: object): Type;
-  // TODO function forTypes(types, opts)
-  function isType(arg: any): boolean; // TODO
+
+// TODO more specific types than `any`
+export class Type {
+  constructor(schema: any, opts: any);
+  clone(val: any, opts?: any): any;
+  compare(val1: any, val2: any): number;
+  compareBuffers(buf1: any, buf2: any): number;
+  createResolver(type: any, opts?: any): any;
+  decode(buf: any, pos?: any, resolver?: any): any;
+  encode(val: any, buf: any, pos?: any): any;
+  equals(type: any): any;
+  fingerprint(algorithm?: any): any;
+  fromString(str: any): any;
+  readonly aliases: any;
+  readonly name: string|undefined;
+  readonly branchName: string|undefined;
+  readonly typeName: string;
+  inspect(): string;
+  isValid(val: any, opts?: any): any;
+  random(): Type;
+  schema(opts?: any): any;
+  toJSON(): string;
+  toString(val?: any): any;
+  wrap(val: any): any;
+  fromBuffer(buffer: Buffer, resolver: any, noCheck: boolean): Type; // TODO
+  toBuffer(value: object): Buffer;
+  static forSchema(schema: Schema, opts?: any): Type;
+  static forTypes(types: any, opts?: any): Type;
+  static forValue(value: object, opts?: any): Type;
+  static isType(arg: any): boolean;  // TODO remaining args
 }
 
-export namespace Service {
-  interface Protocol {
-    [key: string]: any; // TODO object should be further specified
-  }
+// TODO: Can this project remove Protocol from types completely, it's a deprecated export?
+export class Protocol {
+  constructor(name: any, messages: any, types: any, ptcl: any, server: any);
+  createClient(opts: any): any;
+  createEmitter(args: any): any;
+  createListener(args: any): any;
+  createServer(opts: any): any;
+  emit(args: any): any;
+  equals(args: any): any;
+  getMessage(args: any): any;
+  getMessages(args: any): any;
+  getName(args: any): any;
+  getSchema(args: any): any;
+  getType(args: any): any;
+  getTypes(args: any): any;
+  inspect(): string;
+  message(name: any): any;
+  on(args: any): any;
+  subprotocol(args: any): any;
+  type(name: any): any;
+  static compatible(clientSvc: any, serverSvc: any): any;
+  static forProtocol(ptcl: any, opts: any): any;
+  static isService(any: any): any;
+}
+
+export class Service {
+  constructor(name: any, messages: any, types: any, ptcl: any, server: any);
+  createClient(options?: Partial<Service.ClientOptions>): Service.Client;
+  createEmitter(args: any): any;
+  createListener(args: any): any;
+  createServer(options?: Partial<Service.ServerOptions>): Service.Server;
+  emit(args: any): any;
```

Comment:
`emit` and `on` are deprecated and there are a few methods/properties missing (`types`, `type(string)`, etc.). Could update the typings to be in sync with the [API doc](https://github.com/mtth/avsc/wiki/API#class-service), same as you did for the types?